packages/react-form/src/hooks/list/hooks/reducer.ts

```diff
@@ -158,6 +174,29 @@ function reduceList<Item extends object>(
         list: action.payload.list.map(initialListItemState),
       };
     }
+    case 'moveFieldItem': {
+      const newList = [...state.list];
```

Comment:
We shouldn't make an unnecessary copy when an index is invalid and we don't do a move.